packages/python-editor/src/KernelDropdown.tsx

```diff
@@ -16,128 +16,73 @@
-  private updateKernel: Function;
-  private kernelOptionElems: Record<string, any>[];
-
-  /**
-   * Construct a new dropdown widget.
-   */
-  constructor(props: DropDownProps) {
-    super(props);
-    this.state = { kernelSpecs: null };
-    this.updateKernel = this.props.updateKernel;
-    this.kernelOptionElems = [];
-    this.getKernelSPecs();
-  }
-
-  /**
-   * Function: Gets kernel specs and state.
-   */
-  private async getKernelSPecs(): Promise<void> {
-    const specs: KernelSpec.ISpecModels = await this.props.runner.getKernelSpecs();
-    this.filterPythonKernels(specs);
-
-    // Set kernel to default
-    this.updateKernel(specs.default);
-
-    this.createOptionElems(specs);
-    this.setState({ kernelSpecs: specs });
-  }
-
-  /**
-   * Function: Filters for python kernel specs only.
-   */
-  private filterPythonKernels = (specs: KernelSpec.ISpecModels): void => {
-    Object.entries(specs.kernelspecs)
-      .filter(entry => entry[1].language.includes('python') === false)
-      .forEach(entry => delete specs.kernelspecs[entry[0]]);
-  };
-
-  /**
-   * Function: Creates drop down options with available python kernel specs.
-   */
-  private createOptionElems = (specs: KernelSpec.ISpecModels): void => {
-    let i = 0;
-    for (const name of Object.keys(specs.kernelspecs)) {
-      const spec = specs.kernelspecs[name];
-      const displayName = spec.display_name ? spec.display_name : name;
-      const elem = React.createElement(
-        'option',
-        { key: i++, value: name },
-        displayName
-      );
-      this.kernelOptionElems.push(elem);
-    }
-  };
-
-  /**
-   * Function: Handles kernel selection from dropdown options.
-   */
-  private handleSelection = (event: any): void => {
-    const selection: string = event.target.value;
-    this.updateKernel(selection);
-  };
-
-  render(): React.ReactElement {
-    return this.state.kernelSpecs
-      ? React.createElement(
-          HTMLSelect,
-          {
-            className: DROPDOWN_CLASS,
-            onChange: this.handleSelection.bind(this),
-            defaultValue: this.state.kernelSpecs.default
-          },
-          this.kernelOptionElems
-        )
-      : React.createElement('span', null, 'Fetching kernel specs...');
-  }
-}
+// eslint-disable-next-line react/display-name
+const DropDown = forwardRef<ISelect, Props>(({ specs }, select) => {
+  const [selection, setSelection] = useState(specs.default);
+
+  // Note: It's normally best to avoid using an imperative handle if possible.
+  // The better option would be to track state in the parent component and handle
+  // the change events there as well, but I know this isn't always possible
+  // alongside jupyter.
+  useImperativeHandle(select, () => ({
+    getSelection: (): string => selection
+  }));
+
+  return (
+    <select
+      className={KERNEL_SELECT_CLASS}
+      onChange={(e): void => setSelection(e.target.value)}
```

Comment:
We need to update the editor widget with a change in kernel selection, so the runner can start the correct session. For now it always uses the `kernelName` set on init state.